test/integration/fetch-middleware.test.ts

```diff
@@ -188,6 +210,22 @@ describe('fetch middleware', function() {
       })
     })
 
+    describe('on a 404 response', function() {
```

Comment:
What do we want to do when a response comes back without errors but an empty `data` member? Is that valid in the jsonapi spec?

Reply:
As discussed offline - I believe it's valid to the spec but unhelpful for programming, so we are rejecting this like we would a 404. I have also added a spec to ensure `belongsTo` data can still come back as `null`.